plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -496,6 +520,58 @@ private synchronized TableInfo getTableUsingStats(TableInfo info, Collection<Com
         }).collect(toImmutableList()));
     }
 
+    private Map<String, List<Object>> getColumnValues(SchemaTableName tableName, TableInfo info, Map<String, Long> distinctValues, Map<String, Object> minimums, Map<String, Object> maximums)
+    {
+        long schemaDictionarySize = (long) getSchema(tableName.getSchemaName()).properties().getOrDefault(SchemaInfo.MAX_DICTIONARY_SIZE, maxDictionarySize);
+        long maxDictionarySize = (long) info.properties().getOrDefault(TableInfo.MAX_DICTIONARY_SIZE, schemaDictionarySize);
+        if (maxDictionarySize == 0) {
+            return Map.of();
+        }
+        List<ColumnInfo> columns = info.columns();
+        Map<String, Type> types = columns.stream().collect(toMap(ColumnInfo::name, ColumnInfo::type));
+        List<String> dictionaryColumns = distinctValues.entrySet().stream()
+                .filter(entry -> entry.getValue() <= maxDictionarySize)
+                .map(Map.Entry::getKey)
+                .collect(toImmutableList());
+        Map<String, FakerColumnHandle> columnHandles = info.columns().stream()
+                .collect(toImmutableMap(ColumnInfo::name, column -> column.handle().withNullProbability(0)));
```

Comment:
This shouldn't be hard coded to 0, we can get the right value from NUMBER_OF_NON_NULL_VALUES and row count statistics

Reply:
Here we generate allowed values, and they shouldn't contain any nulls.